rfcs/0053-pull-request-workflow.md

```diff
@@ -0,0 +1,107 @@
+---
+feature: pull-request workflow
+start-date: 2019-09-28
+author: Ingolf Wagner (@mrVanDalo)
+co-authors: lassulus (@lassulus)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: 
+---
+
+# Summary
+[summary]: #summary
+
+Pull-Request on GitHub are the main way we improve our code base.
+This Document should settle everything that needs to be done
+to bring code upstream in the
+[Nixpkgs Repository](https://github.com/nixos/nixpkgs/).
+
+# Motivation
+[motivation]: #motivation
+
+Eliminate questions from contributors and maintainers
+about what should be done next and who should do it.
+This is not a new approach, it is more a settlement
+on how we do it now.
+
+# Detailed design
+[design]: #detailed-design
+
+Define all steps of a Pull-Request.
+Use Roles to define responsibilities in every step.
+
+Terms like **SHOULD** and **MUST** are defined in 
+[IETF RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Roles
+[roles]: #roles
+
+Everybody involved in the process of contributing has one or multiple
+of the following roles
+
+* `Contributor` is the person proposing the Pull-Request
+* `Bot` is a bot that provides automated feedback
+* `Reviewer` is any person that reviews the Pull-Request
+  (for example a member of [NixOS/nixpkgs-maintainers](https://github.com/orgs/NixOS/teams/nixpkgs-maintainers))
+* `Merger` is any person with merge privileges
+
+The responsibilities of theses roles are defined in the rest of this RFC.
+
+## States of a Pull-Request
+[state]:#states
+
+This diagram defines all states of a Pull-Request,
+and their transitions to other states.
+
+![pull-request state](0053-pull-request-workflow/pull-request-states.svg)
+
+## Responsibilities and Actions
+[responsibilities]:#responsibilities
+
+The responsibilities of every role is defined by the following diagram:
+
+![pull-request activity](0053-pull-request-workflow/pull-request-roles.svg)
+
+## About Pull-Requests
+
+### Packages
+
+* contributors **SHOULD** evaluate and signal that a Backport is necessary.
+* the [NixOS/backports team](https://github.com/orgs/NixOS/teams/backports)
+  **SHOULD** be pinged in situations that are unclear.
+* the [NixOS/backports team](https://github.com/orgs/NixOS/teams/backports)
+  **MUST** be informed about every Backport.
```

Comment:
Is there a standard way to inform the backports team?

Reply:
Taneb, you can cc them using @ NixOS/backports (without the space) in your PR.

Reply:
I just talked samueldr from the @ NixOS/backports group. I will remove this line, and put a link to this document in there https://gist.github.com/grahamc/c60578c6e6928043d29a427361634df6#what-to-backport

Reply:
Could you just inline the important bits? Linking to a private gist looks a bit weird.